src/main/java/de/tum/in/www1/artemis/service/connectors/localci/buildagent/BuildJobExecutionService.java

```diff
@@ -450,15 +450,18 @@ private Path cloneRepository(VcsRepositoryUri repositoryUri, @Nullable String co
         try {
             // Clone the assignment repository into a temporary directory
             // TODO: use a random value if commitHash is null
-            Repository repository = gitService.getOrCheckoutRepository(repositoryUri, Paths.get(CHECKED_OUT_REPOS_TEMP_DIR, commitHash, repositoryUri.folderNameForRepositoryUri()),
-                    false);
+            Repository repository = buildJobGitService.cloneRepository(repositoryUri,
+                    Paths.get(CHECKED_OUT_REPOS_TEMP_DIR, commitHash, repositoryUri.folderNameForRepositoryUri()));
```

Comment:
(nitpick)
we prefer `Path.of` over `Paths.get`.
```suggestion
                    Path.of(CHECKED_OUT_REPOS_TEMP_DIR, commitHash, repositoryUri.folderNameForRepositoryUri()));
```